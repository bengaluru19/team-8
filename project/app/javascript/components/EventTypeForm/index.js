import React from 'react'
import { Field } from 'redux-form'
import moment from 'moment-timezone'
import R from 'ramda'

import Callout from 'components/Callout'

import s from './main.css'

const capitalize = string => string.charAt(0).toUpperCase() + string.slice(1)

const formatGraphQLErrors = errors =>
  R.map(
    ([field, errors]) => (
      <span key={`${field}-error`}>
        <strong>{capitalize(field)}</strong>: {errors.join(', ')}
        <br />
      </span>
    ),
    R.toPairs(errors)
  )

const renderFieldHelper = ({ input, type, label, className, selectOptions }) => {
  switch (type) {
    case 'input':
    case 'text':
      return <input {...input} type={type} className={className} />
    case 'select':
      return (
        <select {...input} className={className}>
          {selectOptions}
        </select>
      )
    default:
      return <Callout type="error" />
  }
}

const renderField = props => {
  const {
    input,
    label,
    type,
    Custom,
    meta: { touched, error, warning },
    className,
    required,
  } = props
  const fieldInput = renderFieldHelper({ input, type, label, className, selectOptions: props.children })
  return (
    <div>
      <label className={s.label}>
        {label} <span className={s.errorMsg}>{touched && error ? error : '*'}</span>
      </label>
      <div>{Custom ? <Custom {...props} /> : fieldInput}</div>
    </div>
  )
}

const isNoErrors = errors => R.isNil(errors) || R.isEmpty(errors)

const EventTypeForm = ({ handleSubmit, disableSubmit, errors }) => (
  <form className={s.form} onSubmit={handleSubmit}>
    {isNoErrors(errors) ? null : <Callout type="error" message={formatGraphQLErrors(errors)} />}
    <div className={s.inputGroup}>
      <Field label="Title" className={s.field} name="title" component={renderField} type="text" />
    </div>
    <div className={s.inputGroup}>
      <button className={`${s.btn} ${s.primary}`} type="submit" disabled={disableSubmit}>
        Save
      </button>
    </div>
  </form>
)

export default EventTypeForm
