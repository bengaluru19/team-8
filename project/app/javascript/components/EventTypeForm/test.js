import React from 'react'
import EventTypeForm from './'
import renderer from 'react-test-renderer'
import { createStore, combineReducers } from 'redux'
import { reducer as formReducer } from 'redux-form'
import { reduxForm } from 'redux-form'
import { Provider } from 'react-redux'

const rootReducer = combineReducers({ form: formReducer })
const store = createStore(rootReducer)
const WithReduxForm = reduxForm({ form: 'test-event-type' })(EventTypeForm)

const handleSubmit = () => {}

test('loads', () => {
  const component = renderer.create(
    <Provider store={store}>
      <WithReduxForm handleSubmit={handleSubmit} disableSubmit={false} />
    </Provider>
  )
  const tree = component.toJSON()

  expect(tree).toMatchSnapshot()
})
