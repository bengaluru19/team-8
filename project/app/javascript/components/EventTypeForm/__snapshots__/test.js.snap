// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <div
    className="inputGroup"
  >
    <div>
      <label
        className="label"
      >
        Title
         
        <span
          className="errorMsg"
        >
          *
        </span>
      </label>
      <div>
        <input
          className="field"
          name="title"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
  <div
    className="inputGroup"
  >
    <button
      className="btn primary"
      disabled={false}
      type="submit"
    >
      Save
    </button>
  </div>
</form>
`;
